scheduler.go

```diff
@@ -32,6 +37,12 @@ func NewScheduler(loc *time.Location) *Scheduler {
 	}
 }
 
+// SetMaxConcurrentJobs limits how many jobs can be running at the same time.
+// This is useful when running resource intensive jobs and a precise start time is not critical.
+func (s *Scheduler) SetMaxConcurrentJobs(n int) {
```

Comment:
```suggestion
func (s *Scheduler) SetMaxConcurrentJobs(n int) error {
	if n < 1 {
		return ErrConcurrentJobNegative
	}
	s.sem = semaphore.NewWeighted(int64(n))
	return nil
}
```

Reply:
Just accepting a uint is probably cleaner here.